File: python_sdk/src/reality_capture/specifications/reconstruction.py
from pydantic import BaseModel, Field
from typing import Optional
from reality_capture.specifications.tiling import ModelingReference, TilingOptions
from reality_capture.specifications.production import Export, ExportCreate


class ReconstructionInputs(BaseModel):
    scene: str = Field(description="Reality data ID (cloud) or local path (on-premise) of ContextScene to process")
    region_of_interest: Optional[str] = Field(description="Path to region of interest file (bucket path in cloud, local path on-premise), used for tiling region of interest",
                                              alias="regionOfInterest", default=None)
    extent: Optional[str] = Field(None,
                                  description="Path to region of interest file (bucket path in cloud, local path on-premise), used for export extent")
    modeling_reference: Optional[str] = Field(None, description="Reality data ID (cloud) or local path (on-premise) of modeling reference to process",
                                              alias="modelingReference")
    presets: Optional[list[str]] = Field(default=None, description="List of paths to preset")
    crs_data: Optional[str] = Field(default=None,
                                    description="Path to CRS data file (bucket path in cloud, local path on-premise).",
                                    alias="crsData")


class ReconstructionOutputs(BaseModel):
    modeling_reference: Optional[ModelingReference] = Field(None, description="Modeling reference",
                                                            alias="modelingReference")
    exports: Optional[list[Export]] = Field(None, description="List of exports")


class ReconstructionOutputsCreate(BaseModel):
    modeling_reference: Optional[bool] = Field(None, description="Modeling reference", alias="modelingReference")
    exports: Optional[list[ExportCreate]] = Field(None, description="Exports")


class ReconstructionSpecificationsCreate(BaseModel):
    inputs: ReconstructionInputs = Field(description="Inputs")
    outputs: ReconstructionOutputsCreate = Field(description="Outputs")
    options: Optional[TilingOptions] = Field(None, description="Options")


class ReconstructionSpecifications(BaseModel):
    inputs: ReconstructionInputs = Field(description="Inputs")
    outputs: ReconstructionOutputs = Field(description="Outputs")
    options: Optional[TilingOptions] = Field(None, description="Options")


class ReconstructionCost(BaseModel):
    gpix: float = Field(description="Number of GigaPixels in the overall inputs.", ge=0)
    mpoints: float = Field(description="Number of MegaPoints in the overall inputs.", ge=0)

File: typescript_sdk/src/reality_capture/specifications/reconstruction.ts
import { z } from "zod";
import { ModelingReference, TilingOptions } from "./tiling";
import { Export, ExportCreate } from "./production";

// ReconstructionInputs
export const ReconstructionInputsSchema = z.object({
  scene: z.string().describe("Reality data id of ContextScene to process"),
  regionOfInterest: z
    .string()
    .regex(/^bkt:.+/)
    .describe(
      "Path in the bucket to region of interest file, used for tiling region of interest"
    )
    .optional(),
  extent: z
    .string()
    .regex(/^bkt:.+/)
    .describe(
      "Path in the bucket to region of interest file, used for export extent"
    )
    .optional(),
  modelingReference: z
    .string()
    .describe("Reality data id of modeling reference to process")
    .optional(),
  presets: z
    .array(z.string())
    .describe("List of paths to preset")
    .optional()
});
export type ReconstructionInputs = z.infer<typeof ReconstructionInputsSchema>;

// ReconstructionOutputs
export const ReconstructionOutputsSchema = z.object({
  modelingReference: ModelingReference.schema
    .optional()
    .describe("Modeling reference"),
  exports: z
    .array(Export.schema)
    .optional()
    .describe("List of exports")
});
export type ReconstructionOutputs = z.infer<typeof ReconstructionOutputsSchema>;

// ReconstructionOutputsCreate
export const ReconstructionOutputsCreateSchema = z.object({
  modelingReference: z
    .boolean()
    .optional()
    .describe("Modeling reference"),
  exports: z
    .array(ExportCreate.schema)
    .optional()
    .describe("Exports")
});
export type ReconstructionOutputsCreate = z.infer<typeof ReconstructionOutputsCreateSchema>;

// ReconstructionSpecificationsCreate
export const ReconstructionSpecificationsCreateSchema = z.object({
  inputs: ReconstructionInputsSchema.describe("Inputs"),
  outputs: ReconstructionOutputsCreateSchema.describe("Outputs"),
  options: TilingOptions.schema
    .optional()
    .describe("Options")
});
export type ReconstructionSpecificationsCreate = z.infer<typeof ReconstructionSpecificationsCreateSchema>;

// ReconstructionSpecifications
export const ReconstructionSpecificationsSchema = z.object({
  inputs: ReconstructionInputsSchema.describe("Inputs"),
  outputs: ReconstructionOutputsSchema.describe("Outputs"),
  options: TilingOptions.schema
    .optional()
    .describe("Options")
});
export type ReconstructionSpecifications = z.infer<typeof ReconstructionSpecificationsSchema>;

// ReconstructionCost
export const ReconstructionCostSchema = z.object({
  gpix: z
    .number()
    .min(0)
    .describe("Number of GigaPixels in the overall inputs."),
  mpoints: z
    .number()
    .min(0)
    .describe("Number of MegaPoints in the overall inputs.")
});
export type ReconstructionCost = z.infer<typeof ReconstructionCostSchema>;